src/engine.rs

```diff
@@ -182,6 +184,25 @@ impl Reedline {
         }
     }
 
+    /// Get a new history session id based on the current time and the first commit datetime of reedline
+    fn create_history_session_id() -> Option<HistorySessionId> {
+        //Sun Feb 28 22:42:07 2021 +1300
+        let first_commit_date_time_naive = NaiveDate::from_ymd(2021, 2, 28).and_hms(22, 42, 7);
+        let first_commit_date_time_utc =
+            DateTime::<Utc>::from_utc(first_commit_date_time_naive, Utc);
```

Comment:
What's the purpose of using the first commit of reedline as the baseline rather than using the standard unix timestamp (i.e. time since Jan 1 1970)?

Reply:
Hey Dan, The purpose was that I wasn't sure exactly if nanoseconds since UNIX_EPOCH would always (in the near future) fit into a i64. So, while I was thinking about that, I just said to myself, I really don't care that it's some universal time thing like unix epoch, I just care that the number is pretty unique and won't overrun an i64 any time soon. So, I chose the first commit date and time as the constant to compare against.

Reply:
I wanted to use a [ulid](https://github.com/ulid/spec) but it's alphanumeric and I didn't want to change the datatype in the sqlite db.

Reply:
OK, it looks like a u64 fits 585 years of nanoseconds, and an i64 292 years, so I'd say go for the standard UNIX timestamp.

```
In [12]: ns_per_year = 365 * 24 * 60 * 60 * 1e9

In [13]: (2 ** 64 - 1) / ns_per_year
Out[13]: 584.942417355072

In [14]: (2 ** 63 - 1) / ns_per_year
Out[14]: 292.471208677536
```

Reply:
I wonder why I can't do this math in nutshell. 🤔

Reply:
Anyway, I changed it to unix_epoch.

Reply:
Great, yes I think it makes sense to keep it simple and unsurprising -- users are sure to look at these things from time to time.